packages/user/src/model.ts

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { IStateDB } from '@jupyterlab/statedb';
+import { ISignal, Signal } from '@lumino/signaling';
+import { JSONObject, ReadonlyPartialJSONObject, UUID } from '@lumino/coreutils';
+import * as env from 'lib0/environment';
+
+import { ICurrentUser, IUser, USER } from './tokens';
+import { getAnonymousUserName, getRandomColor } from './utils';
+
+/**
+ * Default user implementation.
+ */
+export class User implements ICurrentUser {
+  private _id: string;
+  private _name: string;
+  private _username: string;
+  private _color: string;
+  private _anonymous: boolean;
+  private _role: IUser.ROLE;
+  private _cursor?: IUser.Cursor;
+
+  private _isReady = false;
+  private _state: IStateDB;
+  private _ready = new Signal<User, boolean>(this);
+  private _changed = new Signal<User, void>(this);
+
+  /**
+   * Constructor of the User class.
+   *
+   * @argument state: IStateDB to store users information.
+   */
+  constructor(state: IStateDB) {
+    this._state = state;
+    this._fetchUser()
+      .then(() => {
+        this._isReady = true;
+        this._ready.emit(true);
+      })
+      .catch(e => console.error(e));
+  }
+
+  /**
+   * User's ID.
+   */
+  get id(): string {
+    return this._id;
+  }
+
+  /**
+   * User's name.
+   */
+  get name(): string {
```

Comment:
It isn't clear what name this is. Is this mean to be full name, first name, etc. My advice would be for us to follow the convention of something like the schema.org person type, which uses `familyName` and `givenName`. That way as we build UI around this we can know how to construct short and long variants.

Reply:
Agree, that is probably the best solution. The reason I removed the `familyName` is that we were not using it on the actual implementation, but I think I can change it and use `givenName` and `familyName`. For example for `Anonymous Metis` Anonymous could be the name and Metis the family name.